service/frontend/middleware/tokenAuth.go

```diff
@@ -0,0 +1,33 @@
+package middleware
+
+import (
+	"crypto/subtle"
+	"fmt"
+	"net/http"
+)
+
+// TokenAuth implements a similar middleware handler like go-chi's BasicAuth middleware but for bearer tokens
+func TokenAuth(realm string, token string) func(next http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			bearer := r.Header.Get("Bearer")
```

Comment:
I think the token should be sent in `Authorization` header not in `Bearer` header.
I.e., `Authorization: Bearer <token>`